src/utils/key.rs

```diff
@@ -179,6 +179,22 @@ impl KeyBundle {
 
                 Ok(encrypt(cipher, &self.key, None, data)?)
             }
+            "rsa" => {
+                let rsa = Rsa::private_key_from_pem(&self.key)?;
+                if data.len() >= rsa.size() as usize {
+                    return Err(RvError::ErrPkiInternal);
+                }
+
+                let mut buf: Vec<u8> = vec![0; rsa.size() as usize];
+
+                if aad.unwrap_or("0".as_bytes())[0] != b'1' {
```

Comment:
This is very ugly. You first designed the `encrypt`/`decrypt` functions for symmetric ciphers *only*, so there is an AAD parameter. But there is no AAD concept in public key algorithms like RSA or SM2. It's very confusing for other people to understand.

So you either change the name of AAD to another name like 'flag' or so, or you need to define a new structure containing different parameters for different cryptographic algorithms.

Reply:
> > or you need to define a new structure containing different parameters for different cryptographic algorithms.
> 
> Or an `enum` may be more suitable

fixed